test/distributed/partition_kernels.cpp

```diff
@@ -0,0 +1,474 @@
+        }
+    }
+
+    void assert_equal(
+        std::unique_ptr<gko::distributed::Partition<local_index_type,
+                                                    global_index_type>>& part,
+        std::unique_ptr<gko::distributed::Partition<local_index_type,
+                                                    global_index_type>>& dpart)
+    {
+        ASSERT_EQ(part->get_size(), dpart->get_size());
+        ASSERT_EQ(part->get_num_ranges(), dpart->get_num_ranges());
+        ASSERT_EQ(part->get_num_parts(), dpart->get_num_parts());
+        ASSERT_EQ(part->get_num_empty_parts(), dpart->get_num_empty_parts());
+        GKO_ASSERT_ARRAY_EQ(
+            gko::make_array_view(
+                this->ref, part->get_num_ranges() + 1,
+                const_cast<global_index_type*>(part->get_range_bounds())),
+            gko::make_array_view(
+                this->exec, dpart->get_num_ranges() + 1,
+                const_cast<global_index_type*>(dpart->get_range_bounds())));
+        GKO_ASSERT_ARRAY_EQ(
+            gko::make_array_view(
+                this->ref, part->get_num_ranges(),
+                const_cast<comm_index_type*>(part->get_part_ids())),
+            gko::make_array_view(
+                this->exec, dpart->get_num_ranges(),
+                const_cast<comm_index_type*>(dpart->get_part_ids())));
+        GKO_ASSERT_ARRAY_EQ(
+            gko::make_array_view(this->ref, part->get_num_ranges(),
+                                 const_cast<local_index_type*>(
+                                     part->get_range_starting_indices())),
+            gko::make_array_view(this->exec, dpart->get_num_ranges(),
+                                 const_cast<local_index_type*>(
+                                     dpart->get_range_starting_indices())));
+        GKO_ASSERT_ARRAY_EQ(
+            gko::make_array_view(
+                this->ref, part->get_num_parts(),
+                const_cast<local_index_type*>(part->get_part_sizes())),
+            gko::make_array_view(
+                this->exec, dpart->get_num_parts(),
+                const_cast<local_index_type*>(dpart->get_part_sizes())));
+    }
+
+    std::ranlux48 rand_engine;
+
+    std::shared_ptr<gko::ReferenceExecutor> ref;
+    std::shared_ptr<gko::EXEC_TYPE> exec;
+};
+
+TYPED_TEST_SUITE(Partition, gko::test::LocalGlobalIndexTypes);
+
+
+TYPED_TEST(Partition, BuildsFromMapping)
+{
+    using local_index_type = typename TestFixture::local_index_type;
+    using global_index_type = typename TestFixture::global_index_type;
+    comm_index_type num_parts = 7;
+    std::uniform_int_distribution<comm_index_type> part_dist{0, num_parts - 1};
+    auto mapping = gko::test::generate_random_array<comm_index_type>(
+        10000, part_dist, this->rand_engine, this->ref);
+    gko::Array<comm_index_type> dmapping{this->exec, mapping};
+
+    auto part = gko::distributed::Partition<
+        local_index_type, global_index_type>::build_from_mapping(this->ref,
```

Comment:
because use many gko::distributed::Partition<local_index_type, global_index_type>, maybe use alias for it